From 92af7beb790ad9181e5a6a89ac52f177354536f2 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Thu, 8 Dec 2011 18:39:28 -0500
Subject: [PATCH] Delete key should delete files

Make the Delete key delete files like in nautilus-2.x and every other
file manager on the planet. Note: the nautilus-3 Ctrl+Delete shortcut
will also continue to work.

https://bugzilla.gnome.org/show_bug.cgi?id=647048
https://bugs.gentoo.org/show_bug.cgi?id=393663
---
 src/nautilus-view.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/nautilus-view.c b/src/nautilus-view.c
index f51d916..0a50994 100644
--- a/src/nautilus-view.c
+++ b/src/nautilus-view.c
@@ -7237,7 +7237,7 @@ static const GtkActionEntry directory_view_entries[] = {
   /* tooltip */                  NULL,
 				 G_CALLBACK (action_rename_select_all_callback) },
   /* name, stock id */         { NAUTILUS_ACTION_TRASH, NULL,
-  /* label, accelerator */       N_("Mo_ve to Trash"), "<control>Delete",
+  /* label, accelerator */       N_("Mo_ve to Trash"), "Delete",
   /* tooltip */                  N_("Move each selected item to the Trash"),
 				 G_CALLBACK (action_trash_callback) },
   /* name, stock id */         { NAUTILUS_ACTION_DELETE, NULL,
@@ -9895,8 +9895,12 @@ nautilus_view_class_init (NautilusViewClass *klass)
 	binding_set = gtk_binding_set_by_class (klass);
 	gtk_binding_entry_add_signal (binding_set, GDK_KEY_Delete, GDK_CONTROL_MASK,
 				      "trash", 0);
+	gtk_binding_entry_add_signal (binding_set, GDK_KEY_Delete, 0,
+				      "trash", 0);
 	gtk_binding_entry_add_signal (binding_set, GDK_KEY_KP_Delete, GDK_CONTROL_MASK,
 				      "trash", 0);
+	gtk_binding_entry_add_signal (binding_set, GDK_KEY_KP_Delete, 0,
+				      "trash", 0);
 	gtk_binding_entry_add_signal (binding_set, GDK_KEY_KP_Delete, GDK_SHIFT_MASK,
 				      "delete", 0);
 }
-- 
1.7.12

